codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -1083,13 +1082,11 @@ public void flush() {
                     return;
                 }
                 try {
-                    // If we are current channelReadComplete(...) call we should just mark this Channel with a flush
-                    // pending. We will ensure we trigger ctx.flush() after we processed all Channels later on and
+                    // If we are currently in the  channelReadComplete(...) call we should just ignore the flush.
+                    // We will ensure we trigger ctx.flush() after we processed all Channels later on and
                     // so aggregate the flushes. This is done as ctx.flush() is expensive when as it may trigger an
                     // write(...) or writev(...) operation on the socket.
-                    if (inFireChannelReadComplete) {
-                        flushPending = true;
-                    } else {
+                    if (!inFireChannelReadComplete) {
                         flush0(ctx);
```

Comment:
This flush looks (pre-existingly) broken. We're currently in `Unsafe` and this calls flush on the child channel. Shouldn't we be flushing the parent channel instead?

Reply:
The flush here in the child channel will eventually call flush on the parent channel, if we are not in `channelReadComplete` for the parent. IIUC the rational being that if we are in `channelReadComplete` for the parent then we are in the process of flushing the parent channel.

Reply:
nm. This is in the codec, not the child channel.